### ISSUE TITLE
Odd errors from no-undef rule

### ISSUE THREAD

**Issue by eliaslfox** [gajus/eslint-plugin-flowtype#123]
So, I have eslint running on a file which I am attempting to declare some types. For some reason eslint only reports errors for types declared with some names and not others. Eslint also dose not report errors if the types are created but not declared as a global type. My file is as follows,

``` js
// @flow

declare type Response = {

}

declare type Debug = {

}

declare type Testing = {

}

type Wat = {

}
```

This is my .eslintrc.js

``` js
module.exports = {
    "parser": "babel-eslint",
    "extends": [
        "plugin:flowtype/recommended"
    ],
    "env": {
        "browser": true,
        "node": true,
        "mocha": true,
        "es6": true,
    },
    "plugins": [
        "flowtype"
    ],
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "rules": {
        "no-console": ["error", { "allow": ["warn", "error"] }],
        "use-isnan": "error",
        "valid-typeof": "error",
        "dot-notation": "error",
        "default-case": "error",
        "array-callback-return": "error",
        "no-alert": "error",
        "no-caller": "error",
        "no-global-assign": "error",
        "no-labels": "error",
        "no-invalid-this": "error",
        "no-new-wrappers": "error",
        "no-param-reassign": "error",
        "no-self-compare": "error",
        "no-with": "error",
        "vars-on-top": "error",
        "no-catch-shadow": "error",
        "no-undefined": "error",
        "brace-style": "error",
        "camelcase": "error",
        "no-bitwise": "error",
        "no-continue": "error",
        "no-inline-comments": "error",
        "arrow-spacing": "error",
        "generator-star-spacing": ["error", {"before": false, "after": true}],
        "no-duplicate-imports": "error",
        "prefer-template": "error",
        "prefer-spread": "error",
        "prefer-rest-params": "error",
        "array-bracket-spacing": ["error", "never"],
        "block-spacing": "error",
        "comma-spacing": ["error", { "before": false, "after": true }],
        "computed-property-spacing": ["error", "always"],
        "func-call-spacing": ["error", "never"],
        "key-spacing": ["error", { "beforeColon": false, "afterColon": true, "mode": "strict" }],
        "keyword-spacing": ["error", { "before": true, "after": true }],
        "object-curly-spacing": ["error", "always"],
        "semi-spacing": ["error", {"before": false, "after": false}],
        "space-before-blocks": "error",
        "space-before-function-paren": ["error", "never"],
        "space-in-parens": ["error", "never"],
        "space-infix-ops": "error",
        "no-undef": 1,
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
            "always"
        ],
    }
}
```

I have eslint version 3.6.1, babel-eslint 7.0.0 and eslint-plugin-flowtype 2.19.0.

All of these dependencies are installed locally, and run with a npm script. I am not passing any command line options to eslint.

I am on a mac book air, OSX 10.11.5.

**Comment by eliaslfox** [comment]
And my errors from eslint are

```
{someStuff}/interfaces/action.js
  7:14  warning  'Testing' is not defined  no-undef

✖ 1 problem (0 errors, 1 warning)
```

**Comment by Leeds-eBooks** [comment]
@eliaslfox This is slightly off-topic but I think may be important: why are you using `declare`? I have just stripped all `declare`s from my type files, after reading [this Twitter conversation](https://twitter.com/lbljeffmo/status/788504721439031296) – and everything works as before. TL;DR: `declare` is not necessary for code you own.

**Comment by gajus** [comment]
Closing as stale.
